src/mdns/mdns_avahi.cpp

```diff
@@ -1041,11 +1042,12 @@ void PublisherAvahi::ServiceSubscription::HandleResolveResult(AvahiServiceResolv
     bool                   resolved   = false;
     int                    avahiError = AVAHI_OK;
 
-    VerifyOrExit(aEvent == AVAHI_RESOLVER_FOUND, avahiError = avahi_client_errno(mPublisherAvahi->mClient));
+    otbrLogResultCode(aEvent, "resolve service reply: protocol %d %s.%s.%s = host %s port %" PRIu16 " flags %d",
```

Comment:
how do we know that `AVAHI_RESOLVER_FOUND` is zero?

Reply:
The assumption of `otbrLogResultCode` is, the only success result code is 0. If we keep comparing the code with enums, we may lose convenience. 

However, it may not be a bad idea to trade in some convenience to make the logic more rigorous: we can change the definition of `otbrLogResultCode` so that it won't automatically print the exit code. Instead, we embed the result code in the formatted string manually every time. We may call that utility `otbrLogEscalateOn`. Any suggestions?